src/components/TodoForm.tsx

```diff
@@ -0,0 +1,86 @@
+import { ChangeEvent, FC, FormEvent, useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+
+import { Todo } from '../types/Todo';
+import { USER_ID } from '../api/todos';
+
+type Props = {
+  onSubmit: (newTodo: Omit<Todo, 'id'>) => Promise<void>;
+  onSetError: (error: string) => void;
+  areAllCompleted: boolean;
+  isLoading: boolean;
+  todos: Todo[];
+  errorMessage: string;
+};
+
+export const TodoForm: FC<Props> = ({
+  onSubmit,
+  onSetError,
+  areAllCompleted,
+  isLoading,
+  todos,
+  errorMessage,
+}) => {
+  const [todoTitle, setTodoTitle] = useState('');
+
+  const titleField = useRef<HTMLInputElement>(null);
+
+  const handleFocus = () => {
+    if (titleField.current) {
+      titleField.current.focus();
+    }
+  };
+
+  useEffect(handleFocus, [todos, errorMessage]);
```

Comment:
Move the `useEffect `to the end of code
Placing `useEffect `at the end improves readability by letting developers first grasp the main logic before seeing side effects. This promotes clearer structure and consistency in the codebase.